app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java

```diff
@@ -0,0 +1,219 @@
+package com.github.wnder.user;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationManager;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+
+public class FirebaseUserDatabase implements UserDatabase{
+
+    private final CollectionReference picturesCollection;
+    private final CollectionReference usersCollection;
+    private final Context context;
+
+    @Inject
+    public FirebaseUserDatabase(Context context){
+        picturesCollection = FirebaseFirestore.getInstance().collection("pictures");
+        usersCollection = FirebaseFirestore.getInstance().collection("users");
+        this.context = context;
```

Comment:
Maybe there are methode for this in the Storage class, not sure though

Reply:
We want to get rid of the storage class so this is normal